### PR TITLE
Fixing Phase-II Patatrack Failures

### PR DESCRIPTION
**Pull request by AdrianoDee**
Fixing failures in 
- tests in TrackingRecHit2DCUDA_t in #36605
- wf 39434.502 due to esconsumes missing in SiPixelPhase2DigiToClusterCUDA #36604

### CODE REVIEW

**Comment by AdrianoDee** [comment]
@makortel this should fix both issues, thanks for spotting and reporting them!

**Comment by AdrianoDee** [comment]
test parameters:

workflows = 39434.501,39434.502,39434.5

**Comment by AdrianoDee** [comment]
type bug-fix

**Comment by AdrianoDee** [comment]
please test

File: CUDADataFormats/TrackingRecHit/test/TrackingRecHit2DCUDA_t.cpp [hunk 1]
```diff
@@ -16,18 +16,25 @@ int main() {
...
-    TrackingRecHit2DHost tkhitH(nHits, nModules, 0, nullptr, nullptr, stream, &tkhit);
+    TrackingRecHit2DGPU tkhitPhase2(nHits, true, 0, nullptr, nullptr, stream);
+    testTrackingRecHit2D::runKernels(tkhitPhase2.view());
+
+    TrackingRecHit2DHost tkhitH(nHits, false, 0, nullptr, nullptr, stream, &tkhit);
     cudaStreamSynchronize(stream);
     assert(tkhitH.view());
     assert(tkhitH.view()->nHits() == unsigned(nHits));
-    assert(tkhitH.view()->nMaxModules() == unsigned(nModules));
+    assert(tkhitH.view()->nMaxModules() == 1856);
```
Comment (slava77) [inline CUDADataFormats/TrackingRecHit/test/TrackingRecHit2DCUDA_t.cpp:31]: shouldn't this be `phase1PixelTopology::numberOfModules`?
Similar for the phase2 case below

**Comment by AdrianoDee** [comment]
please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/36607	5179447 
- code changes are in line with the PR description and the details/discussion in the related issues #36604 and #36605
- jenkins tests pass, including the affected workflows/tests and comparisons with the baseline show no (relevant) differences
    - tests with GPU enabled were not submitted, but the fixes appear correct in the GPU context anyways

**Comment by mmusich** [comment]
urgent

**Comment by mmusich** [comment]
can @cms-sw/heterogeneous-l2 please have a look?

**Comment by fwyzard** [comment]
as soon as I'm no longer sick ...

**Comment by fwyzard** [comment]
thanks @makortel :+1:

**Comment by perrotta** [comment]
+1

**Pull request merged**